oci/client.py

```diff
@@ -268,6 +269,7 @@ def _authenticate(
         self,
         image_reference: typing.Union[str, om.OciImageReference],
         scope: str,
+        retry_on_missing_quota: bool=True,
```

Comment:
how about making this an `int` (amount of remaining retries) + keep it abstract (just: retries).

we can still only do retries for http-429-case for now. however we might add other cases later